torch/distributed/fsdp/flatten_params_wrapper.py

```diff
@@ -339,6 +353,26 @@ def _unflatten_params(self) -> None:
 
         del self.flat_param
 
+    @contextlib.contextmanager
+    def unflatten_params(self) -> Generator:
+        """
+        Unflatten params. If the current instance is already unflattened, then
+        it will remain unflattened after the context manager exits.
+        """
+        orig_flattened = self.flat_param is not None
```

Comment:
Would the code be simpler if we added a check and just returned a noop context manager if orig_flattened = False? Such as

```
orig_flattened = ...
if not orig_flattened:
    return suppress()
# Rest of code
```